dot/network/discovery.go

```diff
@@ -49,9 +54,16 @@ type discovery struct {
 	ds                 *badger.Datastore
 	pid                protocol.ID
 	minPeers, maxPeers int
+
+	metrics map[string]interface{}
 }
 
 func newDiscovery(ctx context.Context, h libp2phost.Host, bootnodes []peer.AddrInfo, ds *badger.Datastore, pid protocol.ID, min, max int) *discovery {
+	metrics.Enabled = true
+	discoveryMetrics := make(map[string]interface{})
+	discoveryMetrics[checkPeerCountMetrics] = metrics.GetOrRegisterGauge(checkPeerCountMetrics, nil)
+	discoveryMetrics[peersStoreMetrics] = metrics.GetOrRegisterGauge(peersStoreMetrics, nil)
```

Comment:
Any reason why this is a `map[string]interface{}`? Could it not be `map[string]metrics.Gauge`?
```suggestion
	discoveryMetrics := map[string]metrics.Gauge{
	           checkPeerCountMetrics: metrics.GetOrRegisterGauge(checkPeerCountMetrics, nil),
	           peersStoreMetrics:  metrics.GetOrRegisterGauge(peersStoreMetrics, nil)
	}
```

Reply:
changed from interface to gauge struct